homeassistant/components/lcn/config_flow.py

```diff
@@ -152,15 +137,22 @@ async def async_step_reconfigure(
         if user_input is not None:
             user_input[CONF_HOST] = reconfigure_entry.data[CONF_HOST]
 
+            if not (user_input.items() <= reconfigure_entry.data.items()):
```

Comment:
why is this check needed?

Reply:
If the user does not change anything and tries to close the reconfiguration dialog, he would get an `already_configured´ message. I thinks that's misleading.
Using this check the dialog closes and the connection is renewed.

Reply:
I just realized, that this part does not work as intended. I'll fix it.

Reply:
Without this check the user would get an `already_configured` error if he does not change the IP/port combination. This would also occur if he changes the other settings.
So we only need the `asnyc_abort_entries_match` if the user changes the IP/port because that could interfere with another already existing ConfigEntry.

Reply:
So wait, you mean that self.async_abort_entries_match triggers on the current entry?

Reply:
Actually, it's the other way around. It should trigger if any other entry already contains the given keys with the same values.
But as far as I remember it also triggers if the data in the current entry does not change (for the two specified keys). But that should be fine.

Reply:
I think we should rather fix this in the config flow than to create such workaround, as this can be quite a common use case

Reply:
I re-checked this. Indeed, when using `_async_abort_entries_match` in a reconfigure flow, it will abort also for the entry which is beeing reconfigured.
I adresses this in a separate PR: #141003 
However, I'm not sure if this PR fixes the issue in the most suitable way since I am not so familiar with the internals of the configuration entries.